doc/decisions/0011-protocol-version-compatibility.md

```diff
@@ -0,0 +1,56 @@
+# 0011. Protocol compatibility between App versions
+
+| Date | Tags |
+|---|---|
+| Fri Oct 05  | protocol, messenger |
+
+## Status
+
+draft
+
+## Context
+
+This document enumerates the rules we have to adhere to when doing any protocol modifications and summarizes best practises to do so.
+In the light of recent issues (code landed in develop branch, producing "invisible" messages for previous app versions), it's best
+to capture it in the ADR for future reference.
+
+## Decision
+
+Rules for making any changes related to message payload (transit fields defined in `status-im.transport.message.transit` ns):
+
+- Every new change must be backward compatible back to released App version `0.9.20` and higher. That means that whenever for example
+  new feature (like message replies recently) is added, we can't of course do nothing with the fact that older versions can't properly
+  display new feature, but we must ensure that at least functionality existing in the older version is not affected at all.
+  So in the example above, even if the older version can't display the fact that some message is reply to other message, it still has to
+  be able to receive and display it as a standalone message.
+  
+- This means we **can't** under any circumstances increment the tag of the existing message type handler (like `c4` -> `c5`) because older
```

Comment:
With PSS old version won't be forward compatible so we'll ditch everything. We should plan ahead to make sure we clean up and think of the future as much as possible, as it will be the last occasion